kythe/go/extractors/gcp/config/converter.go

```diff
@@ -120,7 +110,28 @@ func readConfigFile(input string) (*rpb.Config, error) {
 	return conf, nil
 }
 
-func initializeArtifacts(build *cloudbuild.Build) {
-	build.Artifacts = &cloudbuild.Artifacts{}
-	build.Artifacts.Objects = &cloudbuild.ArtifactObjects{}
+// buildSystemStepper encapsulates the logic for adding extra build steps for a
+// specific type of build system (maven, gradle, etc).
+// TODO(danielmoy): we will almost certainly need to support more fine-grained
+// steps and/or artifacts.  For example now artifacts are prepended, we might
+// need ones that are appended.  We might need build steps that occur before or
+// directly after cloning, but before other common steps.
+type buildSystemStepper interface {
+	// artifactsPre should be prepended to the list of artifacts returned from
+	// the cloudbuild invocation.
+	artifactsPre() []string
+	steps(conf *rpb.ExtractionHint) []*cloudbuild.BuildStep
```

Comment:
Not needed, but since you comment everything else for this type, can you add a comment for this function as well?